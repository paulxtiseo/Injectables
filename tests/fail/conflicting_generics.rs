use injectables::{injectable, inject_fields};

#[injectable]
pub struct GenericBase<T> {
  pub data: T,
}

#[inject_fields(GenericBase<i32>, GenericBase<String>)]  // should fail - conflicting types for 'data'
pub struct ConflictingGenerics {
  pub name: String,
}

fn main() {}