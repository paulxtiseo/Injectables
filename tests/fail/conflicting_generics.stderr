error: Circular injection chain detected: ConflictingGenerics already depends on GenericBase
 --> tests/fail/conflicting_generics.rs:8:1
  |
8 | #[inject_fields(GenericBase<i32>, GenericBase<String>)]  // should fail - conflicting types for 'data'
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `inject_fields` (in Nightly builds, run with -Z macro-backtrace for more info)
